frontend/src/metabase-lib/urls.ts

```diff
@@ -71,7 +71,7 @@ export function getUrlWithParameters(
         clean,
         originalQuestion: question,
         includeDisplayIsLocked,
-        query: { objectId },
+        query: objectId === undefined ? {} : { objectId },
```

Comment:
This needed to be changed because of this change
https://github.com/metabase/metabase/pull/31891/files#diff-6f3fb471d1683163e47bdf80234bcb078ca59c2c6e35ee36718b338d87620a20

Otherwise the question URL would have `?objectId=` in the query params whenever `objectId` is undefined.

Reply:
`objectId` isn't important to the behaviour introduced in the PR. But this code was previously relying on the fact that parameters with null or undefined values would be removed from the query params. If `objectId` is undefined here, then this `query` value would be `{ objectId: undefined }` which wouldn't matter because it would get stripped out of the query params before they became part of the URL. But now that parameters with undefined values should not be stripped out of the URL, this code was causing issues. Without this change every URL returned by `getUrlWithParaameters` would have `?objectId=...` in it.